packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -297,6 +354,11 @@ export default util.createRule({
         return getText(node.expression);
       }
 
+      if (node.object.type === AST_NODE_TYPES.TSNonNullExpression) {
+        // Not supported mixing with TSNonNullExpression
+        return '';
```

Comment:
I'm sorry if this is not that satisfying, considering `TSNonNullExpression` is very common

Reply:
👍 this strategy makes sense to me.